### PR TITLE
ComputeCanada support

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Add instructions for compiling on Cedar and Graham
- Unify installation instructions on HPC systems

Breaking:
- Installation instructions on Wheeler are now different, the old are no longer supported.

I'm requesting reviews from people who use the systems this changes. For @geoffrey4444 I had Wheeler in mind, but please feel free to test any others too if you have access. For @gsb76 I also had Wheeler in mind. @trevor-vincent could you test that things compile and the tests pass on Cedar and Graham? Some might time out on Graham or Cedar, I can't remember which one but one is slower than the other. I'm hoping to get feedback from people who haven't done these setups before to make sure the instructions are clear enough :)

Documentation rendered: http://gh.nilsdeppe.com/spectre/pr_661/

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [ ] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
This has been updated to also *partially* support Niagara. There is a bug in the AXV512F intrinsics header for GCC 7 that needs to be fixed. I've contacted support about this but haven't heard anything back yet. We have the same issue on Stampede2

Update 1 Apr 25, 11:50am EDT: ComputeCanada support responded, they will fix it "as soon as possible"

**Comment by nilsdeppe** [comment]
This now also has support for Niagara. @kidder could you test that?

**Review by mar-celine — DISMISSED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please look at this soon as you can

**Comment by geoffrey4444** [comment]
I'll take a look, but my ComputeCanada account has lapsed. I'll first need to get it reinstated.

**Comment by nilsdeppe** [comment]
@geoffrey4444  I was only asking you to try Wheeler :) but feel free to try CC as well

**Comment by nilsdeppe** [comment]
@kidder could you test this on Niagara?

**Comment by kidder** [comment]
tested on Niagara

**Review by kidder — DISMISSED** [review]

File: docs/Installation/InstallationOnClusters.md [hunk 1]
```diff
@@ -39,13 +51,13 @@ To run the tests:
...
-   `mkdir $SPECTRE_HOME/build_[gcc|clang] && cd $SPECTRE_HOME/build_[gcc|clang]`
-3. Run `. $SPECTRE_HOME/support/Environments/wheeler_[gcc|llvm].env` to load
-   the GCC or LLVM/Clang environment
-4. Run `cmake -D CMAKE_BUILD_TYPE=[Release|Debug]
-   -D CMAKE_Fortran_COMPILER=gfortran $SPECTRE_HOME`
-5. Run `make -j4`
+Follow the general instructions using `wheeler` for `SYSTEM_TO_RUN_ON`, except
+you do not need to install any dependencies, so you can skip steps 5 and 6. You
+can optionally compile using LLVM/Clang by sourcing `wheeler_llvm.sh` instead of
+`wheeler_gcc.sh`
```
Comment (geoffrey4444) [inline docs/Installation/InstallationOnClusters.md]: Shouldn't the name be wheeler_clang.sh, since everywhere else we call it "clang" and not "llvm"?

**Review by geoffrey4444 — DISMISSED** [review]
I approve but have one question: why is the wheeler environment for clang named wheeler_llvm.sh and not wheeler_clang.sh? Not worth holding up the pull request, but I thought I'd check if you'd prefer using everywhere "clang" for consistency.

**Comment by nilsdeppe** [comment]
Okay, rebased on develop and changed `wheeler_llvm` to `wheeler_clang` :)

**Review by kidder — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**